src/main/java/gregtech/api/capability/impl/DistillationTowerLogicHandler.java

```diff
@@ -0,0 +1,218 @@
+            workable.isOutputsFull = true;
+
+            return false;
+        }
+
+        workable.isOutputsFull = false;
+        if (recipe.matches(true, importInventory, importFluids)) {
+            workable.getMetaTileEntity().addNotifiedInput(importInventory);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Needs to be overriden for multiblocks that have different assemblies than the standard distillation tower.
+     *
+     * @param structurePattern the structure pattern
+     */
+    public void determineLayerCount(@NotNull BlockPattern structurePattern) {
+        this.layerCount = structurePattern.formedRepetitionCount[1] + 1;
+    }
+
+    /**
+     * Needs to be overriden for multiblocks that have different assemblies than the standard distillation tower.
+     */
+    public void determineOrderedFluidOutputs() {
+        // noinspection SimplifyStreamApiCallChains
+        List<MetaTileEntityMultiblockPart> fluidExportParts = tower.getMultiblockParts().stream()
+                .filter(iMultiblockPart -> iMultiblockPart instanceof IMultiblockAbilityPart<?>abilityPart &&
+                        abilityPart.getAbility() == MultiblockAbility.EXPORT_FLUIDS &&
+                        abilityPart instanceof MetaTileEntityMultiblockPart)
+                .map(iMultiblockPart -> (MetaTileEntityMultiblockPart) iMultiblockPart)
+                .collect(Collectors.toList());
+        // the fluidExportParts should come sorted in smallest Y first, largest Y last.
+        List<IFluidHandler> orderedHandlerList = new ObjectArrayList<>();
+        List<IFluidTank> tankList = new ObjectArrayList<>();
+        int firstY = tower.getPos().getY() + 1;
+        int exportIndex = 0;
+        for (int y = firstY; y < firstY + this.layerCount; y++) {
+            if (fluidExportParts.size() <= exportIndex) {
+                orderedHandlerList.add(FakeTank.INSTANCE);
+                tankList.add(FakeTank.INSTANCE);
+                continue;
+            }
+            MetaTileEntityMultiblockPart part = fluidExportParts.get(exportIndex);
+            if (part.getPos().getY() == y) {
+                List<IFluidTank> hatchTanks = new ObjectArrayList<>();
+                // noinspection unchecked
+                ((IMultiblockAbilityPart<IFluidTank>) part).registerAbilities(hatchTanks);
+                if (hatchTanks.size() == 1)
+                    orderedHandlerList.add(FluidTankHandler.getTankFluidHandler(hatchTanks.get(0)));
+                else orderedHandlerList.add(new FluidTankList(false, hatchTanks));
+                tankList.addAll(hatchTanks);
+                exportIndex++;
+            } else if (part.getPos().getY() > y) {
+                orderedHandlerList.add(FakeTank.INSTANCE);
+                tankList.add(FakeTank.INSTANCE);
+            } else {
+                GTLog.logger.error("The Distillation Tower at " + tower.getPos() +
```

Comment:
This log message can be parametrized: `error("first half {} second half", tower.getPos())`